tmt/steps/__init__.py

```diff
@@ -558,8 +568,17 @@ def wake(self, keys: Optional[List[str]] = None) -> None:
                 # FIXME Enable type check once StepData defined more precisely
                 self.data[key] = value  # type: ignore
 
-    def go(self, *args: Any, **kwargs: Any) -> None:
-        """ Go and perform the plugin task """
+    # NOTE: it's tempting to rename this method to `go()` and use more natural
+    # `super().go()` in child classes' `go()` methods. But, `go()` does not have
+    # the same signature across all plugin types, therefore we cannot have shared
+    # `go()` method in superclass - overriding it in (some) child classes would
+    # raise a linter error reporting superclass signature differs from the one
```

Comment:
`linter` -> `typing linter` would mabe be more descriptive

Reply:
Fixed.